R/shrd_helpers_levels.R

```diff
@@ -0,0 +1,103 @@
+#' Finds the indices of big segments that are close ratio_median-wise
+#'
+#' @description
+#' Starting from the largest segment, we find big segments if their ratio_median
+#' difference is less than the the given threshold. We save their indices.
+#'
+#' @param segments_copy A data frame containing segment data. For the first pass, these segments
+#' are those with size >= 3mb.
+#' @param thr A float: the threshold previously estimated via KDE. Used to determine whether
+#' big segments are merged.
+#'
+#' @export
+
+FindBigSegmentsIndices <- function(segments_copy, thr) {
+  # Getting largest segment by size
+  largest_segment_index = segments_copy[which.max(segments_copy[,7]), 1]
+  largest_segment_ratio = segments_copy[which.max(segments_copy[,7]), 6]
+  largest_segment_ratio = largest_segment_ratio + 0.00001
+
+  # Getting segment index whose ratio_median is closest to largest_segment
+  closest_segment_index = segments_copy[which.min(abs(segments_copy[,6] - largest_segment_ratio)),1]
+
+  segments_indices = c(largest_segment_index, closest_segment_index)
+
+  #Finding segments with ratio_medians close to segments in segments_indices
+  N = dim(segments_copy)[1]
+  for (i in 1:N) {
+    num_of_times_close_to_segments = 0
+    n = length(segments_indices)
+    # Iterate through segments in segments_indices. We compare segment i against all of these
+    # We add +1 to num every time segment i and the other segment have a ratio_median difference
+    # less than the threshold
+    for (j in 1:n) {
+      if (abs(segments_copy[i,6] - segments_copy[which((segments_copy[,1] == segments_indices[j])),6]) < thr) {
+        num_of_times_close_to_segments = num_of_times_close_to_segments + 1
+      }
+    }
+
+    # If segment i is close to every single segment in the list, then we add segment i
+    # to the list too.
+    if (num_of_times_close_to_segments == n) {
+      segments_indices = c(segments_indices, segments_copy[i,1])
+    }
+  }
+
+  segments_indices = unique(segments_indices) # make sure there's no repeats
+  segments_indices
+}
+
+#' TODO: write docs
+ReadSegmFile<-function(seg_file_name){
+  aa<-scan(seg_file_name,nlines=1,what="character",quiet = T)
+  k<-1
+
+  while((length(grep("#",aa))!=0)&&(grep("#",aa)==1)){
+    aa<-scan(seg_file_name,skip=k,nlines=1,what="character",quiet = T)
+    k<-k+1
+  }
+
+  tmp<-read.delim(seg_file_name,skip=k-1, header=T,sep="\t")
+  tmp<-data.matrix(tmp)
+  if(k==2){
+    results<-getInfoSegm(scan(seg_file_name,nlines=1,what="character",quiet = T))
```

Comment:
should this be GetInfoSegm?

Reply:
Looks like it, yup
How did this function run though when testing? o.0

Reply:
As discussed, this is part of `ReadSegmFile` which is called by `PrepForLevelsInitialization` on an intermediate file generated during process. However, this part of the function is never actually executed because the if condition isn't satisfied. The creation of the intermediate file is not necessary, so I will create an issue to refactor the code to do everything in-memory.